### ISSUE TITLE
Publishing of draft releases is broken

### ISSUE THREAD

**Issue by antonu17** [concourse/github-release-resource#70]
After upgrade to v3.9.0 (and v3.9.1) publishing of draft release is failing.
Actually, draft release is creating, but subsequent `get` operation fails with following message:
```
error running command: GET https://api.github.com/repos/myuser/myrepo/git/refs/tags/mydrafttag: 404 Not Found []
```
The same pipeline works well on v3.8.0. Looks like it was probably broken by #64

**Comment by antonu17** [comment]
I've just tried to build docker image from 0c66a411eec390abb1ed03dab753287487e54039, export `rootfs.tar.gz` from the image, put the archive on my 3.9.1 worker instance to `/opt/concourseci/worker/3.9.1/assets/resources/github-release/` and extract it to `/opt/concourseci/worker/3.9.1/assets/resource-images/github-release/rootfs`. After this dirty hack my pipeline started working with draft releases on concourse v3.9.1, so it's definitely was broken in #64

**Comment by evanfarrar** [comment]
👍  I saw this behavior on wings env in the BBL pipeline. Our workflow is to publish a draft release and then have a PM (me) do release notes before publishing.

**Comment by DanielJonesEB** [comment]
We're also experiencing this problem in our pipelines on the implicit `get`, which in our particular case is leading to some pain as some `put`s succeed, then this one fails, which causes external state to be inconsistent.

Distributed transactions for resources, anyone? (That is _very_ much a joke.)

```
error running command: GET https://api.github.com/repos/engineerbetter/concourse-up/git/refs/tags/0.8.4: 404 Not Found []
```

**Comment by shubhaat** [comment]
We also ran into this in the routing-team, our workflow is similar to what @evanfarrar mentioned. We publish a draft, update the release notes, and then publish. If a user is subscribed to the repo they get an automated email when a release is published, and this would go out with no release notes if we don't follow this workflow.

**Comment by mltang64** [comment]
BOSH Windows is affected by this issue as well. We took a look at PR #72 and it looks good to us. Do you have any indication of when this issue might be resolved?

**Comment by aaronshurley** [comment]
routing-team here again. We would love to see this get resolved as we prefer to publish drafts for review before publicly notifying all subscribers

**Comment by antonu17** [comment]
Hi @mltang64 @aaronshurley @vito. 
I'm not sure why this PR is not reviewed/merged yet, I think it's just a matter of time.
Tell me if I have to do anything else to get #72 PR merged?

**Comment by shubhaat** [comment]
routing-team here again. As Aaron mentioned above, we prefer to publish drafts before publishing a final release so we can update the release notes before publishing a final. With this issue blocking the way, we currently do this in our manually, which is non ideal.  It would really help us to have this issue resolved.

**Comment by vito** [comment]
Fixed by #72.
